lib/components/key-handler.js

```diff
@@ -149,20 +150,38 @@ export function keyToggleHandler({keyCode, keyName, keyEventName}: DecoratorProp
       }
 
       renderDecoratedComponent(): ReactElement {
-        const {keyCode, keyName} = this.state;
+        const {keyValue, keyCode, keyName} = this.state;
 
-        return React.createElement(component, { ...this.props, keyCode, keyName });
+        return React.createElement(component, { ...this.props, keyValue, keyCode, keyName });
       }
 
       handleToggleKey(event: KeyboardEvent): void {
-        if (this.state.keyCode === event.keyCode) {
-          return this.setState({ keyCode: null, keyName: null });
+        if (!checkKeyValues(event, this.state)) {
+          return this.setState({ keyValue: null, keyCode: null, keyName: null });
         }
 
-        this.setState({ keyCode: event.keyCode, keyName: keyNames(event.keyCode) });
+        this.setState({ keyValue: event.key || keyNameVals(keyNames(event.keyCode)), keyCode: event.keyCode, keyName: keyNames(event.keyCode) });
       }
     };
   };
 }
 
+/**
+ * Helpers
+ */
+function valFromCode(keyCode: ?number): ?string {
+  if (!keyCode) {
+    return;
+  }
+
+  return keyNameVals(keyNames(keyCode));
+}
+
+function checkKeyValues(event: KeyboardEvent, obj: Props | State): boolean {
```

Comment:
I don't like the name `obj` here, but there's no real need to name this really, you can destruct `obj` here like this: 

```js
function checkKeyValues(event: KeyboardEvent, {keyValue, keyCode, keyName}: Props | State): boolean {
  // ...
}
```

PS: I like that this function is inside this file vs in `utils.js`